sdk/tables/azure-data-tables/azure/data/tables/_table_batch.py

```diff
@@ -581,27 +260,31 @@ def upsert(self, entity: EntityType, mode: Union[str, UpdateMode] = UpdateMode.M
                 :caption: Creating and adding an entity to a Table
         """
         self._verify_partition_key(entity)
-        temp = entity.copy()  # type: ignore
-
-        partition_key = _prepare_key(temp["PartitionKey"])
-        row_key = _prepare_key(temp["RowKey"])
-        temp = _add_entity_properties(temp)
-
-        if mode == UpdateMode.MERGE:
-            self._batch_merge_entity(
+        entity = _add_entity_properties(entity)
```

Comment:
Do you want to change the order here? We've decided to investigate in the coder work: https://github.com/Azure/azure-sdk-for-python/pull/31862#discussion_r1347841711

Reply:
It doesn't look like this was added to the encoder issue thread? #26318
I will revert this change and add a comment - but it needs to be tracked as well.